### PR TITLE
LPS-144032 Test Autom. Different conditions from the same filter combined

### PR DESCRIPTION
**Pull request by Luizcarlosqueiroz**
**Ticket**:
[LPS-144032](https://issues.liferay.com/browse/LPS-144032)
[LPS-124960](https://issues.liferay.com/browse/LPS-124960)

**Motivation**:
Test automation: 'Different conditions from the same filter combined by AND (only for categories and tags)'

### CODE REVIEW

**Comment by Luizcarlosqueiroz** [comment]
@CarlosBrichete @Tim-Cao

**Comment by Tim-Cao** [comment]
I will have a look at today. Thanks @Luizcarlosqueiroz

File: portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro [hunk 1]
```diff
@@ -26,6 +26,40 @@ definition {
 		Button.clickPublish();
 	}
 
+	macro filterByAssetType {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro:29]: filterByType

File: portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro [hunk 2]
```diff
@@ -26,6 +26,40 @@ definition {
...
+
+		Click(locator1 = "ContentDashboard#FILTER_BY_OPTION");
+
+		WaitForLiferayEvent.initializeLiferayEventLog();
+
+		SelectFrame.selectFrameNoLoading(locator1 = "IFrame#IFRAME");
+
+		Portlet.expandTree();
+
+		for (var typeName : list "${typeNames}") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro:46]: for (var typeName : list "${typeNameList}") {

File: portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro [hunk 3]
```diff
@@ -151,15 +185,15 @@ definition {
...
-
-		KeyPress(
-			locator1 = "ContentDashboard#SEARCH_FIELD",
-			value1 = "\ENTER");
+		if (!(isSet(authorsNames))) {
+			var authorsNames = "Test Test";
+		}
 
-		Click(locator1 = "ContentDashboard#AUTHOR_ELEMENT_CHECKBOX");
+		for (var authorName : list "${authorsNames}") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro:192]: for (var authorName : list "${authorNameList}") {

File: portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro [hunk 4]
```diff
@@ -26,6 +26,40 @@ definition {
...
+	macro filterByAssetType {
+		if (IsElementPresent(locator1 = "ContentDashboard#FILTER_AND_ORDER_DROPDOWN")) {
+			AssertClickNoError(
+				locator1 = "ContentDashboard#FILTER_AND_ORDER_DROPDOWN",
+				value1 = "Filter and Order");
+		}
+
+		var filterOption = "Type";
+
+		Click(locator1 = "ContentDashboard#FILTER_BY_OPTION");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro:38]:
```suggestion
Click(
        filterOption = "Type",
        locator1 = "ContentDashboard#FILTER_BY_OPTION");
```

File: portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro [hunk 5]
```diff
@@ -26,6 +26,40 @@ definition {
...
 	}
 
+	macro filterByAssetType {
+		if (IsElementPresent(locator1 = "ContentDashboard#FILTER_AND_ORDER_DROPDOWN")) {
+			AssertClickNoError(
+				locator1 = "ContentDashboard#FILTER_AND_ORDER_DROPDOWN",
+				value1 = "Filter and Order");
+		}
+
+		var filterOption = "Type";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro:36]: Inline

File: portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro [hunk 6]
```diff
@@ -26,6 +26,40 @@ definition {
...
+			Click(
+				key_nodeName = "${typeName}",
+				locator1 = "Treeview#NODE_ITEM");
+		}
+
+		SelectFrame(value1 = "relative=top");
+
+		AssertClickNoError(
+			locator1 = "Button#SELECT",
+			value1 = "Select");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro:56]:
```suggestion
Button.clickSelect();
```

File: portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro [hunk 7]
```diff
@@ -151,15 +185,15 @@ definition {
...
 		Click(locator1 = "ContentDashboard#SEARCH_FIELD");
 
-		Type(
-			locator1 = "ContentDashboard#SEARCH_FIELD",
-			value1 = "Test");
-
-		KeyPress(
-			locator1 = "ContentDashboard#SEARCH_FIELD",
-			value1 = "\ENTER");
+		if (!(isSet(authorsNames))) {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro:188]: authorNameList

File: portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro [hunk 8]
```diff
@@ -151,15 +185,15 @@ definition {
...
+			var authorsNames = "Test Test";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro:189]: authorNameList

File: portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro [hunk 9]
```diff
@@ -187,13 +221,13 @@ definition {
...
 
-		Click(locator1 = "ContentDashboard#SEARCH_CATEGORY_FIELD");
+		Portlet.expandTree();
 
-		Type(
-			locator1 = "ContentDashboard#SEARCH_CATEGORY_FIELD",
-			value1 = "${categoryName}");
-
-		Click(locator1 = "ContentDashboard#CATEGORY_NAME_ELEMENT");
+		for (var categoryName : list "${categoriesNames}") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/ContentDashboard.macro:226]: for (var categoryName : list "${categoryNameList}") {

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Icon.path [hunk 10]
```diff
@@ -325,6 +325,11 @@
 	<td>//span[*[name()='svg'][contains(@class,'icon-question')]]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>TREE_FILTER_CLEAR</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Icon.path:329]: No usage

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/contentdashboard/ContentDashboard.path [hunk 11]
```diff
@@ -290,7 +290,7 @@
 	<td></td>
 </tr>
 <tr>
-	<td>SEARCH_CATEGORY_FIELD</td>
+	<td>SEARCH_FIELD_MODAL</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/contentdashboard/ContentDashboard.path:293]: No usage so deprecate it

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 12]
```diff
@@ -1658,21 +1658,96 @@ definition {
 	}
 
 	@description = "LPS-124960 When there's a WebContent1 authored by Author1 and has Category1 and Category2 and a WebContent2 authored by Author2 and has Category1, and the user filters by Author1 and Category1 and Category2, only the WebContent1 is shown in the list"
-	@ignore = "true"
 	@priority = "3"
 	test FilterByCategoriesAndAuthor {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1662]: Add **property custom.properties = "jsonws.web.service.paths.excludes=";** to enable JSONUser.setFirstPassword

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 13]
```diff
@@ -1658,21 +1658,96 @@ definition {
...
+				title = "Web Content Title 1");
+		}
+
+		task ("Disable option to strangers to verify email") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Configuration",
+				panel = "Control Panel",
+				portlet = "Instance Settings");
+
+			SystemSettings.gotoConfiguration(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1706]: PortalSettings.gotoConfiguration

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 14]
```diff
@@ -1658,21 +1658,96 @@ definition {
...
+				categoriesNames = "Category1,Category2",
+				filterOption = "Categories");
+		}
+
+		task ("Filter by Author: Test test") {
+			ContentDashboard.filterByAuthorList(filterOption = "Author");
+		}
+
+		task ("Assert if only one asset is displayed") {
+			AssertElementPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1743]: LexiconTable.viewTableEntryTitle

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 15]
```diff
@@ -1658,21 +1658,96 @@ definition {
...
+				key_tableEntry = "Web Content Title 1",
+				locator1 = "LexiconTable#TABLE_ENTRY");
 
+			AssertElementNotPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1747]: LexiconTable.viewTableEntryNotPresent

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 16]
```diff
@@ -1719,30 +1794,249 @@ definition {
 	}
 
 	@description = "LPS-124960 When there's a WebContent1 authored by Author1 and has Category1 and Category2 and a WebContent2 authored by Author2 and has Category1, and the user filters by Author1 and Category1, only the WebContent1 is shown in the list"
-	@ignore = "true"
 	@priority = "3"
 	test FilterByCategoryAndAuthor {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1798]: Add property custom.properties = "jsonws.web.service.paths.excludes="; to enable JSONUser.setFirstPassword

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 17]
```diff
@@ -1719,30 +1794,249 @@ definition {
...
+				title = "Web Content Title 1");
+		}
+
+		task ("Disable option to strangers to verify email") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Configuration",
+				panel = "Control Panel",
+				portlet = "Instance Settings");
+
+			SystemSettings.gotoConfiguration(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1842]: PortalSettings.gotoConfiguration

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 18]
```diff
@@ -1719,30 +1794,249 @@ definition {
...
+				categoriesNames = "Category1",
+				filterOption = "Categories");
+		}
+
+		task ("Filter by Author: Test test") {
+			ContentDashboard.filterByAuthorList(filterOption = "Author");
+		}
+
+		task ("Assert if only one asset is displayed") {
+			AssertElementPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1879]: LexiconTable.viewTableEntryTitle

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 19]
```diff
@@ -1719,30 +1794,249 @@ definition {
...
+				key_tableEntry = "Web Content Title 1",
+				locator1 = "LexiconTable#TABLE_ENTRY");
 
+			AssertElementNotPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1883]: LexiconTable.viewTableEntryNotPresent

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 20]
```diff
@@ -1719,30 +1794,249 @@ definition {
...
+				key_tableEntry = "Web Content Title 2",
+				locator1 = "LexiconTable#TABLE_ENTRY");
+		}
 	}
 
 	@description = "LPS-124960 When there's a WebContent1 authored by Author1 and has Category1 and Category2 and a WebContent2 authored by Author2 and has Category1, and the user filters by Author1, Author2 and Category1, both web content are shown in the list"
-	@ignore = "true"
 	@priority = "3"
 	test FilterByCategoryAndAuthors {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1891]: Add property custom.properties = "jsonws.web.service.paths.excludes="; to enable JSONUser.setFirstPassword

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 21]
```diff
@@ -1719,30 +1794,249 @@ definition {
...
+				title = "Web Content Title 1");
+		}
+
+		task ("Disable option to strangers to verify email") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Configuration",
+				panel = "Control Panel",
+				portlet = "Instance Settings");
+
+			SystemSettings.gotoConfiguration(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1935]: PortalSettings.gotoConfiguration

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 22]
```diff
@@ -1719,30 +1794,249 @@ definition {
...
+
+		task ("Filter by Author: Test test and userfn userln") {
+			ContentDashboard.filterByAuthorTable(
+				authorsNames = "Test,userfn",
+				filterOption = "Author");
+		}
+
+		task ("Assert if both assets are displayed") {
+			for (var assetName : list "Web Content Title 1,Web Content Title 2") {
+				AssertElementPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1975]: LexiconTable.viewTableEntryTitle

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 23]
```diff
@@ -1979,6 +2273,59 @@ definition {
...
+			ContentDashboard.goToContentDashboard();
+		}
+
+		task ("Filter assets by both structure types") {
+			ContentDashboard.filterByAssetType(typeNames = "Basic Web Content,WC Structure Name");
+		}
+
+		task ("Assert if both assets are displayed") {
+			for (var assetName : list "Web Content Title 1,Web Content Title 2") {
+				AssertElementPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:2322]: LexiconTable.viewTableEntryTitle

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 24]
```diff
@@ -1719,30 +1794,249 @@ definition {
...
+		}
+
+		task ("Filter by Tag: tagnumber1") {
+			ContentDashboard.filterByTag(
+				filterOption = "Tags",
+				tagName = "tagnumber1");
+		}
+
+		task ("Assert if only one asset is displayed") {
+			AssertElementPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:2032]: LexiconTable.viewTableEntryTitle

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 25]
```diff
@@ -1719,30 +1794,249 @@ definition {
...
+				key_tableEntry = "Web Content Title 1",
+				locator1 = "LexiconTable#TABLE_ENTRY");
+
+			AssertElementNotPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:2036]: LexiconTable.viewTableEntryNotPresent

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 26]
```diff
@@ -1979,6 +2273,59 @@ definition {
 		}
 	}
 
+	@description = "LPS-124960 When there's a WebContent1 of type Subtype1 and WebContent2 of type Subtype2, and the user filters by Subtype1 and Subtype2, both web content are shown in the list"
+	@priority = "3"
+	test FilterBySubtypeFromDifferentAssets {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:2278]: FilterByType

**Review by Tim-Cao — COMMENTED** [review]
Hi @Luizcarlosqueiroz, would you mind updating according to comments? Thanks!
cc @CarlosBrichete

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase (hunk 26 above)
Comment (Luizcarlosqueiroz) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:2278]: In this case, we already have a test called "FilterByType". The difference between them if that one filters only one type, and the one being developed filters multiple types. My suggestion is to name it "FilterByMultipleTypes"

**Comment by Luizcarlosqueiroz** [comment]
Hey @Tim-Cao ! Updated. Could you check if there's any more to be made?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 27]
```diff
@@ -2005,6 +2289,57 @@ definition {
 		}
 	}
 
+	@description = "LPS-124960 When there's a WebContent1 of type Subtype1 and WebContent2 of type Subtype2, and the user filters by Subtype1 and Subtype2, both web content are shown in the list"
+	@priority = "3"
+	test FilterByMultipleTypes {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:2294]: Would you mind changing to FilterByTypes to align with FilterByCategories, FilterByCategoryAndAuthors, and so on?

**Review by Tim-Cao — COMMENTED** [review]
Hi @Luizcarlosqueiroz, there is only one request left. Could you have a look at? Thanks

**Comment by Luizcarlosqueiroz** [comment]
> Hi @Luizcarlosqueiroz, there is only one request left. Could you have a look at? Thanks

Hey Tim! Done.

**Comment by Tim-Cao** [comment]
Good job @Luizcarlosqueiroz. LGTM!

**Comment by Tim-Cao** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**